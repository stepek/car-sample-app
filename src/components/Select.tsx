import RCSelect from "rc-select"

import React from "react"
import {createUseStyles} from "react-jss"

export interface Options {
  [value: string]: string
}

interface SelectProps {
  placeholder?: string
  options?: Options
  value?: string
  onSelect?: (value: string) => void
}

const useStyles = createUseStyles(theme => ({
  select: {
    fontSize: "1.5em",
    fontFamily: "Arial, sans-serif",
    border: "solid 1px #313131",
    minHeight: "1em",
    "& .rc-select-selector": {},
    "& .rc-select-selection-search > input": {
      width: 0,
    },
    marginBottom: "0.5em",
  },
  dropdown: {
    background: "#ffffff",
    fontFamily: "Arial, sans-serif",
    border: "solid 1px #313131",
    "& .rc-select-item-option-active": {
      background: "#93adff",
    },
    "& .rc-select-item-option-selected": {
      background: "#808cb8",
    },
    "& .rc-select-item-option, & .rc-select-item-empty": {
      padding: 15,
      "& .rc-select-item-option-state": {
        display: "none",
      },
    },

    "&&.rc-select-dropdown-hidden": {
      display: "none",
    },
  },
}))

export function Select({
  options = {},
  value,
  placeholder,
  onSelect,
}: SelectProps) {
  const classes = useStyles()

  return (
    <RCSelect
      className={classes.select}
      value={value}
      dropdownClassName={classes.dropdown}
      placeholder={placeholder}
      autoClearSearchValue={false}
      showSearch={false}
      onSelect={onSelect}
    >
      {Object.entries(options).map(([value, label]) => (
        <RCSelect.Option key={value} value={value} title={label}>
          {label}
        </RCSelect.Option>
      ))}
    </RCSelect>
  )
}

export default Select
