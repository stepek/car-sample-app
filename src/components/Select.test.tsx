import React from "react"
import renderer from "react-test-renderer"

import Select from "./Select"

test("<Select> render correctly", () => {
  const component = renderer.create(<Select placeholder={"Foo Bar Baz"} />)

  expect(component.toJSON()).toMatchSnapshot()
})

test("<Select> render correctly with options", () => {
  const component = renderer.create(
    <Select options={{foo: "FOO", bar: "BAR"}} />,
  )
  expect(component.toJSON()).toMatchSnapshot()
})
