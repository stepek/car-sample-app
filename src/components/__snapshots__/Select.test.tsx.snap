// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select> render correctly 1`] = `
<div
  className="rc-select select-0-2-1 rc-select-single rc-select-show-arrow"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="rc-select-selector"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span
      className="rc-select-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autoComplete="off"
        className="rc-select-selection-search-input"
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        readOnly={true}
        role="combobox"
        style={
          Object {
            "opacity": 0,
          }
        }
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      className="rc-select-selection-placeholder"
    >
      Foo Bar Baz
    </span>
  </div>
  <span
    aria-hidden={true}
    className="rc-select-arrow"
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitUserSelect": "none",
        "userSelect": "none",
      }
    }
    unselectable="on"
  >
    <span
      className="rc-select-arrow-icon"
    />
  </span>
</div>
`;

exports[`<Select> render correctly with options 1`] = `
<div
  className="rc-select select-0-2-1 rc-select-single rc-select-show-arrow"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
>
  <div
    className="rc-select-selector"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span
      className="rc-select-selection-search"
    >
      <input
        aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
        aria-autocomplete="list"
        aria-controls="rc_select_TEST_OR_SSR_list"
        aria-haspopup="listbox"
        aria-owns="rc_select_TEST_OR_SSR_list"
        autoComplete="off"
        className="rc-select-selection-search-input"
        id="rc_select_TEST_OR_SSR"
        onChange={[Function]}
        onCompositionEnd={[Function]}
        onCompositionStart={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        readOnly={true}
        role="combobox"
        style={
          Object {
            "opacity": 0,
          }
        }
        type="search"
        unselectable="on"
        value=""
      />
    </span>
    <span
      className="rc-select-selection-placeholder"
    />
  </div>
  <span
    aria-hidden={true}
    className="rc-select-arrow"
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitUserSelect": "none",
        "userSelect": "none",
      }
    }
    unselectable="on"
  >
    <span
      className="rc-select-arrow-icon"
    />
  </span>
</div>
`;
